src/Microsoft.PowerShell.Commands.Utility/commands/utility/select-object.cs

```diff
@@ -337,6 +337,11 @@ private void ProcessExpressionParameter()
             if (ExcludeProperty != null)
             {
                 _exclusionFilter = new MshExpressionFilter(ExcludeProperty);
+                if ((Property == null) || (Property.Length == 0))
+                {
```

Comment:
> I think you mean that it's very unintuitive 😕 

Under "intuitive" I mean that we use the cmdlet 10 years "as is" without many complaints (In other words, users get results they expect). 
All completely different if you look on the inner side of the cmdlet. I guess it was one of the first cmdlets of Powershell. Today such a design would be immediately rejected. The code is tricky and any chip change results in multiple test crush.
And we absolutely cannot add parameter sets (and moreover we cannot add new exceptions) because this is _breaking change_. 
As I said above, time to replace this cmdlet to a new one with stringent modern design, better UX  and better performance.